CoroutineTemplate/app/src/test/java/co/nimblehq/coroutine/ui/screens/home/HomeFragmentTest.kt

```diff
@@ -0,0 +1,45 @@
+package co.nimblehq.coroutine.ui.screens.home
+
+import co.nimblehq.coroutine.databinding.FragmentHomeBinding
+import co.nimblehq.coroutine.test.TestNavigatorModule.mockMainNavigator
+import co.nimblehq.coroutine.test.getPrivateProperty
+import co.nimblehq.coroutine.test.replace
+import co.nimblehq.coroutine.ui.BaseFragmentTest
+import dagger.hilt.android.testing.*
+import io.kotest.matchers.nulls.shouldNotBeNull
+import io.kotest.matchers.shouldBe
+import io.mockk.mockk
+import org.junit.*
+
+@HiltAndroidTest
+class HomeFragmentTest : BaseFragmentTest<HomeFragment, FragmentHomeBinding>() {
+
+    private val mockViewModel = mockk<HomeViewModel>(relaxed = true)
+
+    @get:Rule
+    var hiltRule = HiltAndroidRule(this)
+
+    @Before
+    fun setup() {
+        hiltRule.inject()
+    }
+
+    @Test
+    fun `When initializing HomeFragment, its views display the text correctly`() {
+        launchFragment()
+        fragment.binding.btNext.text.toString() shouldBe "Next"
+        fragment.binding.btCompose.text.toString() shouldBe "Jetpack Compose"
```

Comment:
I'm curious what's the different between the UI Test and Fragment test in terms of the display? Do we usually do checking the text at the UI Test level?

Reply:
We talked about this many times before as Robolectric can do some UI assertions quickly, and faster than executing a UI test on an emulator. So in case we need to do an integration test on multiple screens or some interactions need real device context, let's do it in UI tests. Otherwise, if we just need to verify some UI element attributes or their interactions on a single Fragment, let's do it with Robolectric as FragmentTest.

Reply:
Thanks @luongvo for helping me explain it 🙇

Reply:
So the conclusion is we will do both, right? @luongvo 🤔

Reply:
@lydiasama Both, but not duplicate the test cases.

> So in case we need to do an integration test on multiple screens or some interactions need real device context, let's do it in UI tests. Otherwise, if we just need to verify some UI element attributes or their interactions on a single Fragment, let's do it with Robolectric as FragmentTest.

I see you guys are trying to set up a UI test runner with the emulator on each individual fragment. I can't get the benefit of it as we can achieve those test cases on a single fragment easily by Robolectric without an emulator. IMO, we should only do UI test with emulator when do need to test an integration flow between multiple screens. 🙏

Reply:
@luongvo Nope. The UI Test runner on each individual fragment in DM is just for flexible to test flow by flow. We're not trying to do the UI Test every page like the Fragment Unit test. For example, The Send Money flow that need to start from the HomeFragment.

Reply:
@lydiasama that sounds right to me 👍